test/lsp/protocol_test_corpus.cc

```diff
@@ -922,4 +922,45 @@ TEST_CASE_FIXTURE(ProtocolTest, "DidChangeConfigurationNotificationUpdatesHighli
                              });
 }
 
+TEST_CASE_FIXTURE(ProtocolTest, "OverloadedStdlibSymbolWithMonkeyPatches") {
+    const bool supportsMarkdown = false;
+    const bool supportsCodeActionResolve = true;
+    auto initOptions = make_unique<SorbetInitializationOptions>();
+    initOptions->supportsSorbetURIs = true;
+    assertErrorDiagnostics(initializeLSP(supportsMarkdown, supportsCodeActionResolve, move(initOptions)), {});
+
+    assertErrorDiagnostics(send(*openFile("monkey_patch.rb", "# typed: false\n"
+                                                             "\n"
+                                                             "module ::Kernel\n"
+                                                             "  def open(*args); end\n"
+                                                             "  module_function :open\n"
```

Comment:
does the problem only reproduce with module_function?

Reply:
Looks like it, I'll simplify the test.